engine/src/main/java/org/hibernate/search/analyzer/impl/AnalyzerReference.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.analyzer.impl;
+
+import org.apache.lucene.analysis.Analyzer;
+
+/**
+ * Reference an analayzer implementation.
+ * <p>
+ * It can be a reference to an {@link Analyzer} or a {@link RemoteAnalyzer}.
+ *
+ * @author Davide D'Alto
+ */
+public class AnalyzerReference {
```

Comment:
I am wondering why you chose this delegation approach instead of an interface and two implementations. That's more what I head in mind:

    interface AnalyzerReference {
        String getName(); // if that's needed here at all?
        void close();
    }

    class LuceneAnalyzerReference implements AnalyzerReference {
        private Analyzer luceneAnalyzer;

        Analyzer getAnalyzer() { ... }

        ...
    }

    // Could live in the ES module, returned by the provider impl there
    class RemoteAnalyzerReference implements AnalyzerReference {
        ...
    }

The advantage being, that a given reference either can be one *or* the other, this becomes more apparent by having a dedicated type for each. Also you can work with the specific implementation type if you are sure of the type in a given context.